src/common/config.h

```diff
@@ -26,74 +21,22 @@ class Config {
   typedef YAML::Node YamlNode;
 
   Config(const std::string options,
-         ConfigMode mode = ConfigMode::training,
-         bool validate = false) {
-    std::vector<std::string> sargv;
-    utils::Split(options, sargv, " ");
-    int argc = (int)sargv.size();
-
-    std::vector<char*> argv(argc);
-    for(int i = 0; i < argc; ++i)
-      argv[i] = const_cast<char*>(sargv[i].c_str());
-
-    initialize(argc, &argv[0], mode, validate);
-  }
+         cli::mode mode = cli::mode::training,
+         bool validate = false);
 
   Config(int argc,
          char** argv,
-         ConfigMode mode = ConfigMode::training,
-         bool validate = true) {
-    initialize(argc, argv, mode, validate);
-  }
-
-  void initialize(int argc,
-                  char** argv,
-                  ConfigMode mode = ConfigMode::training,
-                  bool validate = true) {
-    auto parser = ConfigParser(argc, argv, mode, validate);
-    config_ = parser.getConfig();
-    devices_ = parser.getDevices();
+         cli::mode mode = cli::mode::training,
+         bool validate = true);
 
-    createLoggers(this);
-
-    if(get<size_t>("seed") == 0)
-      seed = (size_t)time(0);
-    else
-      seed = get<size_t>("seed");
-
-    if(mode != ConfigMode::translating) {
-      if(boost::filesystem::exists(get<std::string>("model"))
-         && !get<bool>("no-reload")) {
-        try {
-          if(!get<bool>("ignore-model-config"))
-            loadModelParameters(get<std::string>("model"));
-        } catch(std::runtime_error&) {
-          LOG(info, "[config] No model configuration found in model file");
-        }
-      }
-    } else {
-      auto model = get<std::vector<std::string>>("models")[0];
-      try {
-        if(!get<bool>("ignore-model-config"))
-          loadModelParameters(model);
-      } catch(std::runtime_error&) {
-        LOG(info, "[config] No model configuration found in model file");
-      }
-    }
-    log();
-
-    if(has("version"))
```

Comment:
Is this version info still printed? (please just make sure that it is)

Reply:
Yes, it is, and we have a regression test for this